File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/[C#]-Visual Studio 2010 SDK Samples/Code Sweep/C#,C++/TDD/Mocks/MockWebBrowsingService.cs
/***************************************************************************

Copyright (c) Microsoft Corporation. All rights reserved.
This code is licensed under the Visual Studio SDK license terms.
THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

***************************************************************************/

using System;
using Microsoft.VisualStudio;
using Microsoft.VisualStudio.Shell.Interop;

namespace Microsoft.Samples.VisualStudio.CodeSweep.UnitTests
{
    class MockWebBrowsingService : IVsWebBrowsingService
    {
        public class NavigateEventArgs : EventArgs
        {
            public readonly string Url;
            public NavigateEventArgs(string url)
            {
                Url = url;
            }
        }

        public event EventHandler<NavigateEventArgs> OnNavigate;

        #region IVsWebBrowsingService Members

        public int CreateExternalWebBrowser(uint dwCreateFlags, VSPREVIEWRESOLUTION dwResolution, string lpszURL)
        {
            throw new Exception("The method or operation is not implemented.");
        }

        public int CreateWebBrowser(uint dwCreateFlags, ref Guid rguidOwner, string lpszBaseCaption, string lpszStartURL, IVsWebBrowserUser pUser, out IVsWebBrowser ppBrowser, out IVsWindowFrame ppFrame)
        {
            throw new Exception("The method or operation is not implemented.");
        }

        public int CreateWebBrowserEx(uint dwCreateFlags, ref Guid rguidPersistenceSlot, uint dwId, string lpszBaseCaption, string lpszStartURL, IVsWebBrowserUser pUser, out IVsWebBrowser ppBrowser, out IVsWindowFrame ppFrame)
        {
            throw new Exception("The method or operation is not implemented.");
        }

        public int GetFirstWebBrowser(ref Guid rguidPersistenceSlot, out IVsWindowFrame ppFrame, out IVsWebBrowser ppBrowser)
        {
            throw new Exception("The method or operation is not implemented.");
        }

        public int GetWebBrowserEnum(ref Guid rguidPersistenceSlot, out IEnumWindowFrames ppenum)
        {
            throw new Exception("The method or operation is not implemented.");
        }

        public int Navigate(string lpszURL, uint dwNaviageFlags, out IVsWindowFrame ppFrame)
        {
            if (OnNavigate != null)
            {
                OnNavigate(this, new NavigateEventArgs(lpszURL));
            }

            ppFrame = new MockWindowFrame();
            return VSConstants.S_OK;
        }

        #endregion
    }
}

File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/[VB]-Visual Studio 2010 SDK Samples/Code Sweep/C++,VB/TDD/Mocks/MockWebBrowsingService.vb
'**************************************************************************

'Copyright (c) Microsoft Corporation. All rights reserved.
'This code is licensed under the Visual Studio SDK license terms.
'THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
'ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
'IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
'PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

'**************************************************************************


Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Microsoft.VisualStudio.Shell.Interop
Imports Microsoft.VisualStudio

Namespace Microsoft.Samples.VisualStudio.CodeSweep.UnitTests
    Friend Class MockWebBrowsingService
        Implements IVsWebBrowsingService
        Public Class NavigateEventArgs
            Inherits EventArgs
            Public ReadOnly Url As String
            Public Sub New(ByVal url As String)
                Me.Url = url
            End Sub
        End Class

        Public Event OnNavigate As EventHandler(Of NavigateEventArgs)

#Region "IVsWebBrowsingService Members"

        Public Function CreateExternalWebBrowser(ByVal dwCreateFlags As UInteger, ByVal dwResolution As VSPREVIEWRESOLUTION, ByVal lpszURL As String) As Integer Implements IVsWebBrowsingService.CreateExternalWebBrowser
            Throw New Exception("The method or operation is not implemented.")
        End Function

        Public Function CreateWebBrowser(ByVal dwCreateFlags As UInteger, ByRef rguidOwner As Guid, ByVal lpszBaseCaption As String, ByVal lpszStartURL As String, ByVal pUser As IVsWebBrowserUser, <System.Runtime.InteropServices.Out()> ByRef ppBrowser As IVsWebBrowser, <System.Runtime.InteropServices.Out()> ByRef ppFrame As IVsWindowFrame) As Integer Implements IVsWebBrowsingService.CreateWebBrowser
            Throw New Exception("The method or operation is not implemented.")
        End Function

        Public Function CreateWebBrowserEx(ByVal dwCreateFlags As UInteger, ByRef rguidPersistenceSlot As Guid, ByVal dwId As UInteger, ByVal lpszBaseCaption As String, ByVal lpszStartURL As String, ByVal pUser As IVsWebBrowserUser, <System.Runtime.InteropServices.Out()> ByRef ppBrowser As IVsWebBrowser, <System.Runtime.InteropServices.Out()> ByRef ppFrame As IVsWindowFrame) As Integer Implements IVsWebBrowsingService.CreateWebBrowserEx
            Throw New Exception("The method or operation is not implemented.")
        End Function

        Public Function GetFirstWebBrowser(ByRef rguidPersistenceSlot As Guid, <System.Runtime.InteropServices.Out()> ByRef ppFrame As IVsWindowFrame, <System.Runtime.InteropServices.Out()> ByRef ppBrowser As IVsWebBrowser) As Integer Implements IVsWebBrowsingService.GetFirstWebBrowser
            Throw New Exception("The method or operation is not implemented.")
        End Function

        Public Function GetWebBrowserEnum(ByRef rguidPersistenceSlot As Guid, <System.Runtime.InteropServices.Out()> ByRef ppenum As IEnumWindowFrames) As Integer Implements IVsWebBrowsingService.GetWebBrowserEnum
            Throw New Exception("The method or operation is not implemented.")
        End Function

        Public Function Navigate(ByVal lpszURL As String, ByVal dwNaviageFlags As UInteger, <System.Runtime.InteropServices.Out()> ByRef ppFrame As IVsWindowFrame) As Integer Implements IVsWebBrowsingService.Navigate
            If OnNavigateEvent IsNot Nothing Then
                RaiseEvent OnNavigate(Me, New NavigateEventArgs(lpszURL))
            End If

            ppFrame = New MockWindowFrame()
            Return VSConstants.S_OK
        End Function

#End Region
    End Class
End Namespace
